src/track/track.cpp

```diff
@@ -698,6 +698,11 @@ void Track::initId(TrackId id) {
     // generated by the Database itself.
 }
 
+void Track::resetId() {
+    QMutexLocker lock(&m_qMutex);
```

Comment:
This doesn't make sense, the id is not protected by the lock. I would recommend to adopt the same strategy as for initId() which is also invoked only from GlobalTrackCache. The ID of a track must only be modified by GlobalTrackCache while locked!

Reply:
I don't understand your point here. The call stacks of initId() and resetId() are very similar both private and both used the lock. We can of cause get rid of the lock in both cases because both are done from the GUI aka library thread and are int which is written atomic. 
What could be improved here?

Reply:
Ups, sorry. I must have had  an earlier state of the code in mind. It's the dirty flag that doesn't need to be set. The locking seems to be correct!